qiskit_ibm/ibm_provider.py

```diff
@@ -0,0 +1,965 @@
+        if any([hub, group, project]) and not all([hub, group, project]):
+            raise IBMProviderValueError('The hub, group, and project parameters must all be '
+                                        'specified when storing a default provider to disk: '
+                                        'hub = "{}", group = "{}", project = "{}"'
+                                        .format(hub, group, project))
+        # If specified, get the provider to store.
+        default_provider_hgp = HubGroupProject(hub, group, project) \
+            if all([hub, group, project]) else None
+        credentials = Credentials(token=token, url=url,
+                                  default_provider=default_provider_hgp, **kwargs)
+        store_credentials(credentials,
+                          overwrite=overwrite)
+
+    @staticmethod
+    def saved_account() -> Dict[str, str]:
+        """List the account saved on disk.
+
+        Returns:
+            A dictionary with information about the account saved on disk.
+
+        Raises:
+            IBMProviderCredentialsInvalidUrl: If invalid IBM Quantum
+                credentials are found on disk.
+        """
+        stored_credentials, _ = read_credentials_from_qiskitrc()
+        if not stored_credentials:
+            return {}
+        credentials = list(stored_credentials.values())[0]
+        if credentials.url != QISKIT_IBM_API_URL:
+            raise IBMProviderCredentialsInvalidUrl(
+                'Invalid IBM Quantum credentials found on disk.')
+        return {
+            'token': credentials.token,
+            'url': credentials.url
+        }
+
+    def backends(
+            self,
+            name: Optional[str] = None,
+            filters: Optional[Callable[[List[IBMBackend]], bool]] = None,
+            **kwargs: Any
+    ) -> List[IBMBackend]:
+        """Return all backends accessible via this provider, subject to optional filtering.
+
+        Args:
+            name: Backend name to filter by.
+            filters: More complex filters, such as lambda functions.
+                For example::
+
+                    IBMProvider.backends(filters=lambda b: b.configuration().n_qubits > 5)
+            kwargs: Simple filters that specify a ``True``/``False`` criteria in the
+                backend configuration, backends status, or provider credentials.
+                An example to get the operational backends with 5 qubits::
+
+                    IBMProvider.backends(n_qubits=5, operational=True)
+
+        Returns:
+            The list of available backends that match the filter.
+        """
+        # pylint: disable=method-hidden
+        # pylint: disable=arguments-differ
+        # This method is only for faking the subclassing of `BaseProvider`, as
+        # `.backends()` is an abstract method. Upon initialization, it is
+        # replaced by a `IBMBackendService` instance.
```

Comment:
This was replaced by `IBMQDeprecatedBackendService`, which was removed in #35. Although that PR should have updated this method to return `self._backend.backends()` instead of keeping this method empty and continuing to use `backends` as an attribute.

Reply:
Fixed!